packages/react/src/components/Auth/Auth.tsx

```diff
@@ -42,6 +43,12 @@ function Auth({
   const [defaultEmail, setDefaultEmail] = useState('')
   const [defaultPassword, setDefaultPassword] = useState('')
 
+  useEffect(() => {
+    if (onViewChange) {
+      onViewChange(authView)
+    }
+  }, [authView, onViewChange])
```

Comment:
You can avoid the useEffect by making a `handleSetAuthView` function and use it instead of `setAuthView`:
```ts
const handleSetAuthView = (view: ViewType) => {
  setAuthView(view);
  onViewChange?.(view);
};
```